etcdserver/api/v3rpc/key.go

```diff
@@ -131,6 +131,13 @@ func (s *kvServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.Co
 		plog.Panic("unexpected nil resp.Header")
 	}
 	s.fillInHeader(resp.Header)
+
+	resprange, err := s.Range(context.TODO(), &pb.RangeRequest{Key: []byte("foo"), Serializable: true})
+	if err != nil {
+		return nil, err
+	}
+	resp.Header.Revision = resprange.Header.Revision
+
```

Comment:
Not sure where else we should populate the Revision. Everywhere with response header?